sql/api/src/main/scala/org/apache/spark/sql/catalyst/trees/QueryContexts.scala

```diff
@@ -165,6 +172,20 @@ case class DataFrameQueryContext(stackTrace: Seq[StackTraceElement]) extends Que
     builder ++= " was called from\n"
     builder ++= callSite
     builder += '\n'
+
+    if (pysparkOriginInfo.nonEmpty) {
+      builder ++= "\n== PySpark call site ==\n"
+      builder ++= "\""
+
+      builder ++= pysparkFragment
+      builder ++= "\""
+      builder ++= " was called from\n"
+      builder ++= pysparkCallSite
+      builder += '\n'
+    }
+
+    PySparkCurrentOrigin.clear()
```

Comment:
it looks tricky to call `clear` inside a `lazy val`...

Reply:
Sounds reasonable. Let me update